sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -840,10 +840,17 @@ abstract class BinaryComparison extends BinaryOperator with Predicate {
 
   final override val nodePatterns: Seq[TreePattern] = Seq(BINARY_COMPARISON)
 
-  override def checkInputDataTypes(): TypeCheckResult = super.checkInputDataTypes() match {
-    case TypeCheckResult.TypeCheckSuccess =>
-      TypeUtils.checkForOrderingExpr(left.dataType, this.getClass.getSimpleName)
-    case failure => failure
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val matched = (left.dataType, right.dataType) match {
+      case (l: DayTimeIntervalType, r: DayTimeIntervalType) => TypeCheckResult.TypeCheckSuccess
```

Comment:
It's a bit weird that we allow different types in binary comparison. Can we fix the type coercion instead? e.g. `TypeCoercion.findTightestCommonType`. This is also more general, we can also support `coalesce(interval1, interval2)`

Reply:
> It's a bit weird that we allow different types in binary comparison. Can we fix the type coercion instead? e.g. `TypeCoercion.findTightestCommonType`. This is also more general, we can also support `coalesce(interval1, interval2)`

Hmm like current?

Reply:
> It's a bit weird that we allow different types in binary comparison. Can we fix the type coercion instead? e.g. `TypeCoercion.findTightestCommonType`. This is also more general, we can also support `coalesce(interval1, interval2)`

If not change here, `checkInputDataTypes` will be false and expression is not resolved for  below UT.
```
 checkEvaluation(EqualTo(
      Literal.create(10,
        YearMonthIntervalType(YearMonthIntervalType.YEAR, YearMonthIntervalType.MONTH)),
      Literal.create(10,
        YearMonthIntervalType(YearMonthIntervalType.MONTH, YearMonthIntervalType.MONTH))), true)
```
If reasonable? or should I just remove the UT